app/models/api_key.rb

```diff
@@ -2,14 +2,18 @@ class ApiKey < ApplicationRecord
   API_SCOPES = %i[index_rubygems push_rubygem yank_rubygem add_owner remove_owner access_webhooks show_dashboard].freeze
   APPLICABLE_GEM_API_SCOPES = %i[push_rubygem yank_rubygem add_owner remove_owner].freeze
 
-  belongs_to :user
+  belongs_to :user, inverse_of: :api_keys
+  belongs_to :owner, polymorphic: true, optional: true
```

Comment:
I wonder if you don't need optional since all keys will be get owners before the validations run. Could then skip the validates on owner too.

Reply:
oh i suppose that's true